pkg/frontend/querymiddleware/codec.go

```diff
@@ -445,26 +446,75 @@ func (prometheusCodec) DecodeLabelsQueryRequest(_ context.Context, r *http.Reque
 	}, nil
 }
 
+// timeParamType enumerates the types of time parameters in Prometheus API.
+// https://prometheus.io/docs/prometheus/latest/querying/api/
+type timeParamType int
+
+const (
+	RFC3339OrUnixMS     timeParamType = iota // <rfc3339 | unix_timestamp> in Prometheus docs
+	DurationMS                               // <duration> in Prometheus docs
+	DurationMSOrFloatMS                      // <duration | float> in Prometheus docs
+)
+
+// promTimeParamDecodable provides common functionality for decoding Prometheus time parameters.
+type promTimeParamDecodable struct {
```

Comment:
isn't this a `decoder`, not a `decodable`? In my mind the query parameter from the request is a decodable.

Reply:
:+1: done